File: outdoor-saas-fe/src/services/planFrameService.ts
/**
 * 方案框架明细服务
 */
import request from './api';
import type { PageResult } from '../types/query';

export interface PlanFrame {
  id: number;
  planId: number;
  frameId: number;
  planCommunityId: number;
  releaseDateBegin?: string;
  releaseDateEnd?: string;
  releaseStatus?: number;
  plan?: {
    id: number;
    planNo: string;
    planName: string;
  };
  frame?: {
    id: number;
    frameNo: string;
    building: string;
  };
  planCommunity?: {
    id: number;
    communityId: number;
  };
}

export interface PlanFrameQueryParam {
  pageNum?: number;
  pageSize?: number;
  planId?: number;
  frameId?: number;
  releaseStatus?: number;
}

export const planFrameService = {
  // 获取所有方案框架明细
  getAll: () => request<PlanFrame[]>('/plan-frame/list'),

  // 根据ID获取
  getById: (id: number) => request<PlanFrame>(`/plan-frame/${id}`),

  // 分页查询（带过滤参数）
  filterPage: (params: PlanFrameQueryParam) =>
    request<PageResult<PlanFrame>>('/plan-frame/filter/page', {
      method: 'POST',
      body: JSON.stringify(params),
    }),

  // 根据方案ID获取
  getByPlan: (planId: number) =>
    request<PlanFrame[]>(`/plan-frame/plan/${planId}`),

  // 根据框架ID获取
  getByFrame: (frameId: number) =>
    request<PlanFrame[]>(`/plan-frame/frame/${frameId}`),

  // 根据方案社区ID获取
  getByPlanCommunity: (planCommunityId: number) =>
    request<PlanFrame[]>(`/plan-frame/plan-community/${planCommunityId}`),

  // 根据发布状态获取
  getByStatus: (releaseStatus: number) =>
    request<PlanFrame[]>(`/plan-frame/status/${releaseStatus}`),

  // 创建
  create: (data: Partial<PlanFrame>) =>
    request<PlanFrame>('/plan-frame', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  // 批量创建
  createBatch: (data: Partial<PlanFrame>[]) =>
    request<PlanFrame[]>('/plan-frame/batch', {
      method: 'POST',
      body: JSON.stringify(data),
    }),

  // 更新
  update: (data: Partial<PlanFrame>) =>
    request<PlanFrame>('/plan-frame', {
      method: 'PUT',
      body: JSON.stringify(data),
    }),

  // 删除
  delete: (id: number) =>
    request<void>(`/plan-frame/${id}`, { method: 'DELETE' }),

  // 批量删除
  deleteBatch: (ids: number[]) =>
    request<void>('/plan-frame/batch', {
      method: 'DELETE',
      body: JSON.stringify(ids),
    }),
};

export default planFrameService;

File: outdoor-saas-be/src/main/java/com/touhuwai/service/PlanFrameService.java
package com.touhuwai.service;

import com.touhuwai.common.PageResult;
import com.touhuwai.entity.PlanFrame;

import java.util.List;

/**
 * 方案框架明细服务接口
 */
public interface PlanFrameService {
    
    /**
     * 根据ID查询方案框架明细
     * @param id 明细ID
     * @return 方案框架明细信息
     */
    PlanFrame getById(Integer id);
    
    /**
     * 查询所有方案框架明细
     * @return 方案框架明细列表
     */
    List<PlanFrame> getAll();
    
    /**
     * 分页查询方案框架明细列表
     * @param planFrame 查询条件
     * @param pageNum 页码
     * @param pageSize 每页大小
     * @return 分页结果
     */
    PageResult<PlanFrame> getPage(PlanFrame planFrame, Integer pageNum, Integer pageSize);
    
    /**
     * 根据方案ID查询明细列表
     * @param planId 方案ID
     * @return 方案框架明细列表
     */
    List<PlanFrame> getByPlanId(Integer planId);
    
    /**
     * 根据框架ID查询明细列表
     * @param frameId 框架ID
     * @return 方案框架明细列表
     */
    List<PlanFrame> getByFrameId(Integer frameId);
    
    /**
     * 根据方案社区ID查询明细列表
     * @param planCommunityId 方案社区ID
     * @return 方案框架明细列表
     */
    List<PlanFrame> getByPlanCommunityId(Integer planCommunityId);
    
    /**
     * 根据发布状态查询明细列表
     * @param releaseStatus 发布状态
     * @return 方案框架明细列表
     */
    List<PlanFrame> getByReleaseStatus(Integer releaseStatus);
    
    /**
     * 新增方案框架明细
     * @param planFrame 方案框架明细信息
     * @return 影响的行数
     */
    int add(PlanFrame planFrame);
    
    /**
     * 批量新增方案框架明细
     * @param list 方案框架明细列表
     * @return 影响的行数
     */
    int batchAdd(List<PlanFrame> list);
    
    /**
     * 更新方案框架明细
     * @param planFrame 方案框架明细信息
     * @return 影响的行数
     */
    int update(PlanFrame planFrame);
    
    /**
     * 根据ID删除方案框架明细
     * @param id 明细ID
     * @return 影响的行数
     */
    int delete(Integer id);
    
    /**
     * 批量删除方案框架明细
     * @param ids ID列表
     * @return 影响的行数
     */
    int batchDelete(List<Integer> ids);
}
